platform/duration.cpp

```diff
@@ -74,12 +69,16 @@ std::string Duration::GetLocalizedString(std::initializer_list<Units> units, Loc
 
 std::string Duration::GetPlatformLocalizedString() const
 {
-  static auto const kCurrentUnitSeparator = GetUnitSeparator(GetCurrentLocale());
-  static auto const kCurrentGroupingSeparator = GetUnitsGroupingSeparator(GetCurrentLocale());
+  // I'm pretty sure that better to traverse 10-elements array than lock a "static" mutex (2 times).
+  auto const & loc = GetCurrentLocale();
+  auto const kCurrentUnitSeparator = GetUnitSeparator(loc);
+  auto const kCurrentGroupingSeparator = GetUnitsGroupingSeparator(loc);
+
```

Comment:
The main idea of using the `static` is to cache the values for the `current locale` as it may be called relatively often (up to 1 sec if the widged/pp of the track recording is open)
With this fix the std::string Duration::GetPlatformLocalizedString() const doesn't need. And we can use only the Locale in the method: 
```cpp
std::string Duration::GetString(std::initializer_list<Units> units, Locale locale)
  auto const kCurrentUnitSeparator = GetUnitSeparator(locale);
  auto const kCurrentGroupingSeparator = GetUnitsGroupingSeparator(locale);
```
and pass the `CurentLocale()` as a default parameter to the method in the .h

I can make this small fix later.

Reply:
I understand that, but static gives more overhead here (can't say 100%, but pretty sure).
If you want more optimizations, calculate and store these separators in Locale once.

Reply:
Reducing unnecessary calculations here will help for longer routes without charger. `Locale.m_groupingSeparator` is for digits, while here it is for separating a digit from a shortened word.

Reply:
Ok, I was wrong and GetCurrentLocale() is definitely not a lightweight function. So keep as-is but 1 static var.